### PR TITLE
Rigil fixes for alpha, solhint cleanup and Synthetix swing trade reduction

### PR DESCRIPTION
**Pull request by justinjmoses**
A few fixes here

1. Cleaned up solhint rules
2. Better VSCode and solhint integration
3. Minor fix to `Synthetix.exchange` to reduce gas (from `6,770,253` to `6,742,558`) 
4. Minor upgrade of `truffle` and `web3`
5. Fixed `solhint` to take quoted glob (so that it actually checks subdirectories correctly)
6. Cleaned up test coverage of SIP-21
7. Fixed deploy script

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,11 +1,6 @@
 {
 	"solidity.compileUsingRemoteVersion": "soljson-v0.4.25+commit.59dbf8f1",
-	"solidity.soliumRules": {
```
Comment (justinjmoses) [inline .vscode/settings.json]: These belong in `.solhint.json` and were moved there.

File: contracts/Depot.sol [hunk 2]
```diff
@@ -131,7 +131,7 @@ contract Depot is SelfDestructible, Pausable {
         // Other contracts needed
         ISynthetix _synthetix,
         ISynth _synth,
-		IFeePool _feePool,
+        IFeePool _feePool,
```
Comment (justinjmoses) [inline contracts/Depot.sol]: Lint fix

File: contracts/Synthetix.sol [hunk 3]
```diff
@@ -478,18 +478,18 @@ contract Synthetix is ExternStateToken {
...
-                sourceCurrencyKey[0] == 0x69 && destinationCurrencyKey[0] == 0x73){
+            if (
+                (sourceCurrencyKey[0] == 0x73 && sourceCurrencyKey != "sUSD" && destinationCurrencyKey[0] == 0x69) ||
+                (sourceCurrencyKey[0] == 0x69 && destinationCurrencyKey != "sUSD" && destinationCurrencyKey[0] == 0x73)
+            ) {
                 // Double the exchange fee and sub the fee from the amountReceived
-                amountReceived = destinationAmount.multiplyDecimal(SafeDecimalMath.unit().sub(exchangeFeeRate.mul(2)));
-            } else {
-                amountReceived = destinationAmount.multiplyDecimal(SafeDecimalMath.unit().sub(exchangeFeeRate));
+                multiplier = 2;
```
Comment (justinjmoses) [inline contracts/Synthetix.sol]: Main change to reduce gas

File: contracts/interfaces/IERC20.sol [hunk 4]
```diff
@@ -23,14 +23,14 @@ contract IERC20 {
     function decimals() public view returns (uint8);
 
     event Transfer(
-      address indexed from,
```
Comment (justinjmoses) [inline contracts/interfaces/IERC20.sol]: lint fixes

File: contracts/interfaces/ISynth.sol [hunk 5]
```diff
@@ -1,9 +1,9 @@
 pragma solidity 0.4.25;
 
 interface ISynth {
-  function burn(address account, uint amount) external;
```
Comment (justinjmoses) [inline contracts/interfaces/ISynth.sol]: lint fixes

File: package.json [hunk 6]
```diff
@@ -9,7 +9,7 @@
 		"coverage": "solidity-coverage",
 		"ganache": "ganache-cli -e 1000000000000 -l 8000000 -t 2018-03-13T00:00:00",
 		"ganache:int": "npm run ganache -- --acctKeys keys.json",
-		"lint": "solhint contracts/**/*.sol && eslint \"**/*.js\"",
+		"lint": "solhint \"contracts/**/*.sol\" && eslint \"**/*.js\"",
```
Comment (justinjmoses) [inline package.json]: This was preventing `solhint` from actually taking the glob (cc @Agupane)

File: truffle.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
 	networks: {
 		development: {
-			host: '127.0.0.1',
+			host: 'localhost',
```
Comment (justinjmoses) [inline truffle.js]: just for consistency

File: contracts/Synthetix.sol (hunk 3 above)
Comment (jacko125) [reply contracts/Synthetix.sol]: @justinjmoses  Looks good the gas opt / reducing complexity

**Pull request merged**